src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/Designer/ILaunchProfileEditorClientSession.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug.Designer
+{
+    /// <summary>
+    /// Proffered by the client, for notifications from the server.
+    /// </summary>
+    internal interface ILaunchProfileEditorClientSession
+    {
+        /// <summary>
+        /// <para>
+        /// Informs the client editor that a profile has been added, including all metadata and values.
+        /// </para>
+        /// <para>
+        /// When the client first connects to the server this method will be called multiple times to inform the client of all existing profiles.
+        /// </para>
+        /// </summary>
+        /// <param name="correlationId">
+        /// The correlation ID provided by the client, if this profile is being added as a result of calling <see cref="ILaunchProfileEditorConnection.AddAsync(Guid, string, string)"/>.
+        /// </param>
+        /// <param name="profile">The newly-added <see cref="Profile"/> along with all metadata and properties.</param>
+        /// <returns>
+        /// A Task that completes when the client has received and scheduled the profile addition.
+        /// </returns>
+        Task LaunchProfileAddedAsync(Guid? correlationId, Profile profile);
```

Comment:
One thought here is that the UI won't know when it has received all of the initial profiles. There are two minor downsides to this:

1. We won't be able to show a "loading" message as it is not possible to differentiate between "no profiles exist" and "we haven't received the first yet" states.
2. If they trickle in while the UI is visible, each arrival will trigger UI updates. We expect only a few profiles (1-5) so this isn't likely to be a huge issue in practice.

A simple change could be to pass `ImmutableArray<Profile>` here, with the assumption that the first call on the client includes the full snapshot. Future calls would likely only include a single item in that array.